torch/csrc/jit/runtime/static/passes.cpp

```diff
@@ -812,18 +812,18 @@ void EliminateTrivialEquallySplit(std::shared_ptr<torch::jit::Graph>& graph) {
       continue;
     }
 
-    Node* list_unpack_node = value_out->uses()[0].user;
-    if (list_unpack_node->kind() != prim::ListUnpack) {
+    Node* user_node = value_out->uses()[0].user;
+    if (user_node->kind() != prim::ListUnpack && user_node->kind() != aten::__getitem__) {
       continue;
     }
```

Comment:
this worries me. 1) what if there is more than one `__getitem__`? as written it looks like we will just leave the other ones around?
2) what if the number of `__getitem__` calls in the input program doesn't match the size of the output list?

Reply:
i changed it to work only fb::equally_split. This has an argument num_splits and I am using that to do all the following checks

1. same as number of uses of this module
2. all the uses are get_item
3. get_item uses is not duplicated

only after this I do the fuse